org.eclipse.Java.json

```diff
@@ -0,0 +1,52 @@
+  "sdk-extensions" : [ "org.freedesktop.Sdk.Extension.openjdk11" ],
+  "command" : "eclipse",
+  "modules" : [ {
+    "name" : "openjdk",
+    "buildsystem" : "simple",
+    "build-commands" : [ "/usr/lib/sdk/openjdk11/installjdk.sh" ]
+  }, {
+    "name" : "eclipse",
+    "buildsystem" : "simple",
+    "build-commands" : [
+      "mv eclipse /app",
+      "mkdir -p /app/bin",
+      "ln -s /app/eclipse/eclipse /app/bin",
+      "install -Dm 644 org.eclipse.Java.desktop /app/share/applications/org.eclipse.Java.desktop",
+      "install -Dm 644 org.eclipse.Java.appdata.xml /app/share/metainfo/org.eclipse.Java.appdata.xml",
+      "mkdir -p /app/share/icons/hicolor/32x32/apps",
+      "if [ -f /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse32.png ] ; then cp -p /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse32.png /app/share/icons/hicolor/32x32/apps/org.eclipse.Java.png ; fi",
+      "mkdir -p /app/share/icons/hicolor/64x64/apps",
+      "if [ -f /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse64.png ] ; then cp -p /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse64.png /app/share/icons/hicolor/64x64/apps/org.eclipse.Java.png ; fi",
+      "mkdir -p /app/share/icons/hicolor/128x128/apps",
+      "if [ -f /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse128.png ] ; then cp -p /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse128.png /app/share/icons/hicolor/128x128/apps/org.eclipse.Java.png ; fi",
+      "mkdir -p /app/share/icons/hicolor/256x256/apps",
+      "if [ -f /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse256.png ] ; then cp -p /app/eclipse/plugins/org.eclipse.epp.package.java_*/eclipse256.png /app/share/icons/hicolor/256x256/apps/org.eclipse.Java.png ; fi",
+      "sed -i -e '/osgi.configuration.area/d' /app/eclipse/eclipse.ini",
+      "echo \"-Dosgi.configuration.area=@user.home/.var/app/org.eclipse.Java/eclipse/configuration\" >> /app/eclipse/eclipse.ini",
+      "echo \"--patch-module=java.base=/app/eclipse/flatpak-dev-shim.jar\" >> /app/eclipse/eclipse.ini"
+    ],
+    "sources" : [ {
+      "type" : "archive",
+      "url" : "https://download.eclipse.org/technology/epp/downloads/release/2019-09/R/eclipse-java-2019-09-R-linux-gtk-x86_64.tar.gz",
+      "sha512" : "eb408902f079d6666863bc318a0586589be9a86e4cd57125ef1f97eb4f4a9d6b70aa52ea23129f5f95eb513c3ce1889683516d91e85a484fcae7328fa8e1eeff",
+      "strip-components" : 0
+    }, {
+      "type" : "file",
+      "path" : "org.eclipse.Java.desktop",
+      "dest-filename" : "org.eclipse.Java.desktop"
+    }, {
+      "type" : "file",
+      "path" : "org.eclipse.Java.appdata.xml",
+      "dest-filename" : "org.eclipse.Java.appdata.xml"
+    } ]
+  },
+  "shim-module.json"
+  ],
+  "finish-args" : [ "--filesystem=host", "--share=network", "--share=ipc", "--socket=x11", "--socket=wayland", "--allow=devel", "--socket=session-bus", "--require-version=1.0.2", "--env=PATH=/app/bin:/app/jdk/bin:/usr/bin" ]
```

Comment:
Nitpick: put this just before the modules.

Reply:
Move the "finish-args"?

Reply:
Yes

Reply:
But then they're not at the finish ;-)

Is there a style guide? In the reference it comes *after* modules: http://docs.flatpak.org/en/latest/flatpak-builder-command-reference.html#flatpak-manifest

Reply:
We don't have a style guide documented somewhere. It makes it easier for us to see those permissions without having to scroll down to the end of the file.

Reply:
Fixed in 1d73c69